charts/redpanda/values.go

```diff
@@ -1363,6 +1363,10 @@ type SecretRef struct {
 	Name             string `json:"name"`
 }
 
+func (sr *SecretRef) IsReferenceValid() bool {
+	return sr != nil && !helmette.Empty(sr.Key) && !helmette.Empty(sr.Name)
```

Comment:
nit: prefer direct string comparisons and/or using `ptr.Deref` over `helmette.Empty`. The former makes the typing of `.Key` and `.Name` much more clear and confident in itself.

Reply:
I found that transpiled code would wrongly handled cases where secret ref is not provided and key is not empty string, but rather nil. 

`testdata/ci/23-aks-tiered-storage-with-creds-values.yaml.tpl.golden` had the following change

```diff
diff --git a/charts/redpanda/testdata/ci/23-aks-tiered-storage-with-creds-values.yaml.tpl.golden b/charts/redpanda/testdata/ci/23-aks-tiered-storage-with-creds-values.yaml.tpl.golden
index b2244521..02480a60 100644
--- a/charts/redpanda/testdata/ci/23-aks-tiered-storage-with-creds-values.yaml.tpl.golden
+++ b/charts/redpanda/testdata/ci/23-aks-tiered-storage-with-creds-values.yaml.tpl.golden
@@ -1393,6 +1393,11 @@ spec:
           value: ${REDPANDA_SAMPLE_LICENSE}
         - name: RPK_CLOUD_STORAGE_AZURE_SHARED_KEY
           value: ${TEST_AZURE_SHARED_KEY}
+        - name: RPK_CLOUD_STORAGE_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              key: null
+              name: null
         - name: RPK_CLOUD_STORAGE_AZURE_CONTAINER
           value: ${TEST_STORAGE_CONTAINER}
         - name: RPK_CLOUD_STORAGE_AZURE_STORAGE_ACCOUNT
```

I can not compare string to nil as our declaration does not declares pointer to string.

Reply:
The `SecretRef` is not nil 😮‍💨 as the `ConfigurationKey` is set to `cloud_storage_access_key`.